src/wp-admin/includes/plugin.php

```diff
@@ -1203,26 +1203,39 @@ function validate_plugin_requirements( $plugin ) {
 	}
 
 	if ( WP_Plugin_Dependencies::has_unmet_dependencies( $plugin ) ) {
-		$dependencies       = WP_Plugin_Dependencies::get_dependencies( $plugin );
-		$unmet_dependencies = array();
+		$dependency_names       = WP_Plugin_Dependencies::get_dependency_names( $plugin );
+		$unmet_dependencies     = array();
+		$unmet_dependency_names = array();
 
-		foreach ( $dependencies as $dependency ) {
+		foreach ( $dependency_names as $dependency => $dependency_name ) {
 			$dependency_file = WP_Plugin_Dependencies::get_dependency_filepath( $dependency );
 
 			if ( false === $dependency_file ) {
-				$unmet_dependencies['not_installed'][] = $dependency;
+				$unmet_dependencies['not_installed'][ $dependency ] = $dependency_name;
+				$unmet_dependency_names[]                           = $dependency_name;
 			} elseif ( is_plugin_inactive( $dependency_file ) ) {
-				$unmet_dependencies['inactive'][] = $dependency;
+				$unmet_dependencies['inactive'][ $dependency ] = $dependency_name;
+				$unmet_dependency_names[]                      = $dependency_name;
 			}
 		}
 
+		$plugins_page  = is_multisite() ? network_admin_url( 'plugins' ) : admin_url( 'plugins' );
+		$error_message = sprintf(
+			/* translators: 1: Plugin name, 2: A comma-separated list of plugin names, 3: The start of a link to the plugins page, 4: The end of the link. */
+			_n(
+				'<strong>Error:</strong> %1$s requires the following plugin to be installed and activated: %2$s. %3$sManage plugins%4$s',
+				'<strong>Error:</strong> %1$s requires the following plugins to be installed and activated: %2$s. %3$sManage plugins%4$s',
+				count( $unmet_dependency_names )
+			),
+			$plugin_headers['Name'],
+			implode( ', ', $unmet_dependency_names ),
+			'<a href="' . esc_url( $plugins_page ) . '">',
+			'</a>'
```

Comment:
Generally, `_n()` should only be used if the amount is also mentioned in the string.

Second, please do not use placeholders for opening and closing tags.

```suggestion
			/* translators: 1: Plugin name, 2: Number of plugins, 3: A comma-separated list of plugin names, 4: Link to the plugins page. */
			_n(
				'<strong>Error:</strong> %1$s requires %2$d plugin to be installed and activated: %3$s. <a href="%4$s>"Manage plugins</a>',
				'<strong>Error:</strong> %1$s requires %2$d plugins to be installed and activated: %3$s. <a href="%4$s>"Manage plugins</a>',
				count( $unmet_dependency_names )
			),
			$plugin_headers['Name'],
			count( $unmet_dependency_names )
			implode( ', ', $unmet_dependency_names ),
			esc_url( $plugins_page )
```